### PR TITLE
Use the service manager to get the service instance

### PR DESCRIPTION
**Pull request by matthias-ronge**

### CODE REVIEW

File: Kitodo/src/main/java/org/kitodo/services/image/ImageService.java [hunk 1]
```diff
@@ -100,4 +103,20 @@ Image getScaledWebImage(URI imageFileUri, double percent) throws IOException {
...
     }
+
+    /**
+     * Return singleton variable of type FilterService.
+     *
+     * @return unique instance of FilterService
+     */
+    public static ImageService getInstance() {
+        if (Objects.equals(instance, null)) {
+            synchronized (FilterService.class) {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/image/ImageService.java]: Why are you synchronizing on `FilterService.class` and not on  `ImageService.class` as you are inside ImageService class?

File: Kitodo/src/main/java/org/kitodo/services/image/ImageService.java [hunk 2]
```diff
@@ -100,4 +103,20 @@ Image getScaledWebImage(URI imageFileUri, double percent) throws IOException {
     Image getSizedWebImage(URI imageFileUri, int pixelWidth) throws IOException {
         return imageManagement.getSizedWebImage(imageFileUri, pixelWidth);
     }
+
+    /**
+     * Return singleton variable of type FilterService.
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/image/ImageService.java]: You are returning an instance of  `ImageService` class, so you should adjust the JavaDoc.

File: Kitodo/src/main/java/org/kitodo/services/image/ImageService.java (hunk 1 above)
Comment (matthias-ronge) [reply Kitodo/src/main/java/org/kitodo/services/image/ImageService.java]: Changed

File: Kitodo/src/main/java/org/kitodo/services/image/ImageService.java (hunk 2 above)
Comment (matthias-ronge) [reply Kitodo/src/main/java/org/kitodo/services/image/ImageService.java]: Changed

Comment (matthias-ronge) [reply Kitodo/src/main/java/org/kitodo/services/image/ImageService.java]: Changed

**Comment by matthias-ronge** [comment]
Can this branch be merged?

File: Kitodo/src/main/java/org/kitodo/services/ServiceManager.java [hunk 3]
```diff
@@ -105,6 +107,12 @@ private void initializeFilterService() {
         }
     }
 
+    private void initializeImareService() {
```
Comment (Kathrin-Huber) [inline Kitodo/src/main/java/org/kitodo/services/ServiceManager.java]: You have a typo here

File: Kitodo/src/main/java/org/kitodo/services/ServiceManager.java [hunk 4]
```diff
@@ -105,6 +107,12 @@ private void initializeFilterService() {
...
+        if (imageServic == null) {
```
Comment (Kathrin-Huber) [inline Kitodo/src/main/java/org/kitodo/services/ServiceManager.java]: typo here, too

File: Kitodo/src/main/java/org/kitodo/services/ServiceManager.java [hunk 5]
```diff
@@ -315,6 +323,11 @@ public LdapServerService getLdapServerService() {
         return ldapServerService;
     }
```
Comment (Kathrin-Huber) [inline Kitodo/src/main/java/org/kitodo/services/ServiceManager.java]: JavaDoc is missing

File: Kitodo/src/main/java/org/kitodo/services/ServiceManager.java [hunk 6]
```diff
@@ -315,6 +323,11 @@ public LdapServerService getLdapServerService() {
...
 
+    public ImageService getImageSerice() {
```
Comment (Kathrin-Huber) [inline Kitodo/src/main/java/org/kitodo/services/ServiceManager.java]: Here also is a typo

File: Kitodo/src/main/java/org/kitodo/services/image/ImageService.java [hunk 7]
```diff
@@ -26,6 +27,7 @@
 public class ImageService {
 
     private ImageManagementInterface imageManagement;
+    private static ImageService instance = null;
```
Comment (Kathrin-Huber) [inline Kitodo/src/main/java/org/kitodo/services/image/ImageService.java]: If you are creating a singleton, the Constructor needs to be private.

File: Kitodo/src/main/java/org/kitodo/services/image/ImageGenerator.java [hunk 8]
```diff
@@ -54,6 +55,8 @@
  */
 public class ImageGenerator implements Runnable {
     private static final Logger logger = LogManager.getLogger(ImageGenerator.class);
+    private final ServiceManager serviceManager = new ServiceManager();
+    private final ImageService imageService = serviceManager.getImageSerice();
```
Comment (Kathrin-Huber) [inline Kitodo/src/main/java/org/kitodo/services/image/ImageGenerator.java]: In line 150 you are still using the constructor instead of the serviceManager

**Comment by matthias-ronge** [comment]
I have made the requested changes, even if GitHub does not show this. Is there anything else I need to do here, or can it be merged the way it is?

**Comment by matthias-ronge** [comment]
Back to the topic: Is there anything else I need to do here, or can it be merged now as it is?

**Pull request merged**